### ISSUE TITLE
setParameter does not respect global scope value.

### ISSUE THREAD

**Issue by Maksims** [playcanvas/engine#2456]
Using setParameter on some meshes/materials, sets it uniform values, but its also overrides global scope value - which is unintended when other meshes/materials have this uniform undefined.

It is useful to set some global uniforms:
```js
// global value
var scopeGlobal = this.app.graphicsDevice.scope.resolve('custom_uniform');
scopeGlobal.setValue(globalValue);
```
And then be able to override them using local setParameter:
```js
// local value
entity.model.meshInstances[0].setParameter('custom_unform', localValue);
```

**Issue here, is once local value is set, it overrides global value, so further draw calls will have it, even if they have no local value defined. Basically leading to uniforms leak to other drawcalls.**

Here is simple project, to demonstrate an issue: https://playcanvas.com/editor/scene/1000034
Only cube has setParameter called, but it also affects sphere.

**Expected result would be: sphere to use global value, set once before.**
Workaround is to find all meshes that need this uniform, and set them global value directly, but this is hard in dynamic scenes and not efficient.

Also `ScopeSpace.prototype.getSubSpace` is not used anywhere within engine, is it redundant?

**Comment by mvaligursky** [comment]
At the moment the concept of a global uniform that can be overwritten by meshInstance does not exist. setParameter on meshInstance allow to overwrite the parameter value set on material, and after the render call this is restored back to value from material.

**Comment by Maksims** [comment]
Yep, that is the current logic. I assumed it worked with global/local scopes before, so even tested with older engines to see if it was a regression. But was surprised to find out, it was actually always like that.

I think it will be a good mechanics, that would even allow to set global uniforms before layer rendering, that way it becomes layer scope.
Or some other way to set default value for uniform that is not defined on specific drawcall.
